src/codeu/chat/server/Server.java

```diff
@@ -123,6 +124,20 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
       }
     });
 
+    // New Interest - A client wants to add a new interest to the back end.
+    this.commands.put(NetworkCode.NEW_INTERESTS_REQUEST,  new Command() {
+      @Override
+      public void onMessage(InputStream in, OutputStream out) throws IOException {
+
+        final Uuid userid = (Uuid.SERIALIZER).read(in);
+        final Uuid interest = (Uuid.SERIALIZER).read(in);
+        final Interests interests = controller.newInterest(Uuid userid, Uuid interest);
```

Comment:
May have steered you wrong a little bit here, but I think that this should be `newInterest(userid, interest)`, you generally don't have the type when you're calling a function, only when you are defining one.